element_facemap/facial_behavior_estimation.py

```diff
@@ -0,0 +1,361 @@
+    class File(dj.Part):
+        definition = """
+        -> master
+        file_path: varchar(255)     # filepath of video, relative to root data directory
+        """
+
+
+@schema
+class RecordingInfo(dj.Imported):
+    definition = """
+    -> VideoRecording
+    ---
+    px_height                 : smallint  # height in pixels
+    px_width                  : smallint  # width in pixels
+    nframes                   : smallint  # number of frames 
+    fps                       : float     # (Hz) frames per second
+    recording_duration = NULL : float     # video duration in seconds
+    recording_time = NULL     : datetime  # Time at the beginning of the recording with respect to the session
+    """
+
+    @property
+    def key_source(self):
+        return VideoRecording & VideoRecording.File
+
+    def make(self, key):
+        file_paths = (VideoRecording & key).fetch('file_path')
+
+        nframes = 0
+        px_height, px_width, fps = None, None, None
+
+        for file_path in file_paths:
+            file_path = (find_full_path(get_facemap_root_data_dir(), file_path)).as_posix()
+            # Open the video
+            with cv2.VideoCapture(file_path) as cap:
+                info = int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT)), int(cap.get(cv2.CAP_PROP_FRAME_WIDTH)), int(cap.get(cv2.CAP_PROP_FPS))
+                if px_height is not None:
+                    assert (px_height, px_width, fps) == info
+                px_height, px_width, fps = info
+                nframes += int(cap.get(cv2.CAP_PROP_FRAME_COUNT))
+
+        self.insert1({
+            **key,
+            'px_height': px_height,
+            'px_width': px_width,
+            'nframes': nframes,
+            'fps': fps,  # Assuming that this keyword is set correctly but it's usually not.
+            'duration': nframes / fps,                     # TODO Something to think about.
+        })
+    
+
+@schema
+class FacemapParamSet(dj.Manual):
+    definition = """
+    # Parameters used to run the Facemap on each Recording
+    -> VideoRecording
+    paramset_idx        :  smallint
+    ---
+    params              : longblob
+    param_set_hash      : uuid          # unique index
+    do_mot_svd=1        : bool
+    do_mov_svd=0        : bool
+    paramset_desc=''    : varchar(128)
+    """
+
+
+@schema
+class FacemapTask(dj.Manual):
+    definition = """
+    # Manual table for defining a processing task ready to be run
+    -> FacemapParamSet
+    ---
+    processing_output_dir   : varchar(255)              # output directory that will keep the facemap results
+    task_mode='load'        : enum('load', 'trigger')   # 'load': load computed analysis results, 'trigger': trigger computation
+    """
+
+    @classmethod
+    def infer_output_dir(cls, video_key,  relative=False, mkdir=False):
+        video_file = (FacemapParamSet * VideoRecording.File & video_key).fetch('file_path')[0]  # Take 1 video file
```

Comment:
Thinh wrote [these lines](https://github.com/datajoint/element-deeplabcut/blob/main/element_deeplabcut/model.py#L390-L392) for deeplabcut, I presume to speed up the process if there's an excessive number of entries. Mirrored in suggestion below.
```suggestion
        video_file = (FacemapParamSet * VideoRecording.File & video_key).fetch('file_path', limit=1)[0]  # Take 1 video file
```

Reply:
That's right! Thank you! Better to fetch only 1 instead of fetching all the file_paths and then taking the first.